protocols/src/ir.rs

```diff
@@ -162,16 +162,10 @@ impl Transaction {
         arg_types
     }
 
-    /// Retrieves the `SymbolId`s of all the output parameters of a transaction,
-    /// returning an `Iterator`
-    pub fn get_output_param_symbols(&self) -> impl Iterator<Item = SymbolId> {
-        self.args.iter().filter_map(|arg| {
-            if arg.dir == Dir::Out {
-                Some(arg.symbol)
-            } else {
-                None
-            }
-        })
+    /// Retrieves all the output parameters of a transaction,
+    /// returning an `Iterator` of `&Arg`'s
+    pub fn get_output_param_symbols(&self) -> impl Iterator<Item = &Arg> {
+        self.args.iter().filter(|arg| arg.dir == Dir::Out)
```

Comment:
Oh, I see. Makes sense, why you chose the `filter_map`. Feel free to go back to that version. (Also, feel free to point out why my suggestions don't work in the future).